### PR TITLE
bump wp-env to 9.7.0, include blocks in syncpack

### LINKED ISSUE #43957: Sync @wordpress/env
Multiple versions of `@wordpress/env` are in use:

```
{
  "8.13.0": [
    "@woocommerce/plugin-woo-ai@0.5.0",
    "@woocommerce/plugin-woocommerce@8.6.0",
    "@woocommerce/plugin-woocommerce-beta-tester@2.3.0",
    "@woocommerce/plugin-woocommerce-docs@1.0.0",
    "compare-perf@0.0.1"
  ],
  "8.2.0": [
    "@woocommerce/block-library@11.8.0-dev"
  ]
}
```

### PR DESCRIPTION
**Pull request by rrennick**
### Changes proposed in this Pull Request:

This PR 
- updates the pinned `@wordpress/env` version to `^9.7.0` in Syncpack
- removes `@wordpress/env` from the excluded packages for the blocks package in Syncpack

Closes #43957 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. In `trunk` run `pnpm list --recursive @wordpress/env` in the root of the repo
2. All packages should be using `^9.7.0`
3. Repeat the above in this branch
4. Run `syncpack list-mismatches`
5. This should not list any packages 
6. 

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
bump wp-env to 9.7.0, include blocks in syncpack
</details>

### CODE REVIEW

**Comment by samueljseay** [comment]
Can I ask why this is being done, specifically why include blocks? 

Syncpack hasn't really been that useful, its tended to just end up locking packages into specific versions of things when they want to upgrade and I don't think blocks needs to be included in syncpack. if there is solid rationale I'm open to hear it, but overall locking down wp dependencies was a mistake.

Packages have different constraints around what they want to do and actually it doesn't make sense for them all to be stuck on the same versions, for example product editor should be able to use bleeding edge packages because they're dealing with bleeding edge APIs.

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -48,7 +48,7 @@ importers:
         version: link:tools/monorepo-utils
       '@wordpress/data':
         specifier: wp-6.0
-        version: 6.6.1(react@17.0.2)
+        version: 6.6.1(react@18.3.1)
```
Comment (kalessil) [inline pnpm-lock.yaml]: Likely the reason for failing JS tests. To confirm, we can keep only changes around the deps (as per changes in package.json files):

```
 '@wordpress/env':
        specifier: ^9.7.0
```

**Review by kalessil — APPROVED** [review]
Pre-approving (see a comment about the reason for failing tests). Followed the testing instructions - all good.

**Comment by rrennick** [comment]
> if there is solid rationale I'm open to hear it, but overall locking down wp dependencies was a mistake.

@samueljseay I wouldn't argue with that given the issue with TS build. However, `@wordpress/env` isn't a package that is included in any of the builds. We have two types of npm packages: utilities (eg. `rimraf`) and libraries (eg. `@wordpress/data`). I think we should work toward having all of the utilities in Syncpack if they are used in more than one local package.

**Comment by samueljseay** [comment]
@rrennick ah apologies I misread the diff. Yep this looks good 👍🏻

**Comment by adimoldovan** [comment]
Synced with trunk and all tests passed. I'll merge this one.

**Comment by JamieMason** [comment]
👋 I'm curious if you could help me understand, why is it that `pinVersion` is used so much? _Are_ you wanting to pin eg [pnpm to _specifically_ `^9.1.0`](https://github.com/woocommerce/woocommerce/blob/6d99ed6a1d73355da9a0aeac5854d3edb06f12bd/.syncpackrc#L11-L19) or do you actually just want to make sure that pnpm always has the same version? As you've mentioned, using pinVersion this much will create a lot of extra work - but maybe it's not needed?

/cc @samueljseay

**Comment by samueljseay** [comment]
@JamieMason Are you saying our use case of pinning versions rather than keeping them up to date seems unusual? 

For the `@wordpress` dependencies, we originally wanted everyone to get the same API in local dev so they would have a better idea of stable APIs that were available. Its a bit of a unique case because what actually gets loaded in production is served up by Wordpress on the window global and may not represent the version you used in dev.

For those dependencies it was specifically a big problem because they did a heavy refactor of their TS types and now we got stuck on those old versions in syncpack. It was a nice idea but sadly thwarted by those issues and we also discovered that different packages actually should probably be allowed to access bleeding edge apis depending on their goals.

Pnpm we pinned very intentionally because we have had so much trouble with how `pnpm-lock` generation breaks or significantly changes how some dependencies are resolved from version to version causing unusual things to break such as TS types or unit tests. Pnpm moves very fast and has multiple times broken our environment from release to release.

I would say in general any other time we are using pinnedVersion its because we haven't been able to keep up with some kind of breaking change in a dependency. As @rrennick mentioned, I think for this `@wordpress/env` dependency it makes sense because we want a single version of the docker environment to run all our e2e tests across the monorepo and probably can't afford the hassle of a minor version change potentially introducing e2e failures.

**Comment by JamieMason** [comment]
Thanks @samueljseay, that all makes perfect sense. I wondered if potentially there was a gap in the syncpack documentation and you _might_ have been using pinVersion when you may not need to, but it sounds like these are all good uses cases of it and there just happen to be quite a lot of them. I was partly hoping there might be some other feature of syncpack you could use that would be a better fit for the uses cases and be less of a maintenance burden. Thanks for feeding back.

There is a new [pnpm catalogs](https://pnpm.io/catalogs) feature by the way which might help with some of these things too.

**Pull request merged**